yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -438,6 +519,15 @@ private[yarn] class YarnAllocator(
     amClient.releaseAssignedContainer(container.getId())
   }
 
+  private def rackLookUp(host: String, conf: Configuration): String = {
+    hostToRackNameCache.get(host) match {
```

Comment:
Have you noticed a need for this caching?  RackResolver should be doing its own caching, so this shoudn't be needed.